src/js/components/DateInput/DateInput.js

```diff
@@ -149,6 +149,20 @@ Use the icon prop instead.`,
       }
     }, [range, schema, textValue, reference, value]);
 
+    // textValue of MaskedInput is controlled.
+    // for uncontrolled forms, ensure the reset event
+    // resets the textValue
+    useEffect(() => {
+      const handleFormReset = (e) => {
+        if (schema && ref.current && e.target.contains(ref.current)) {
+          setTextValue('');
+        }
+      };
+
+      window.addEventListener('reset', handleFormReset);
+      return () => window.removeEventListener('reset', handleFormReset);
```

Comment:
It's a bit of a corner case, but because this expects the event to bubble up, if the Form's onReset stops the event propagation this won't get called. For example I broke this with:
```jsx
<Form onReset={(e) => {
  console.log('MyReset');
  e.stopPropagation();
}}>
```

Reply:
Hmm -- that's a good note, even if a corner case. I was trying to just rely on native HTML events, but I suppose one option would be to fire a custom event within form. and listen for that within DateInput?

Reply:
I adjusted the scope of the event listener to be placed on the form as opposed to the window. I checked with the code you shared (with custom reset function) and it worked.